importer/src/path_canonicalizer.rs

```diff
@@ -263,6 +232,45 @@ fn free_inputs_in_expression_mut<'a>(
     }
 }
 
+fn canonicalize_inside_pil_statement(
+    statement: &mut PilStatement,
+    path: &AbsoluteSymbolPath,
+    paths: &'_ PathMap,
+) {
+    match statement {
+        PilStatement::LetStatement(_, _, type_scheme, e) => {
+            if let Some(type_scheme) = type_scheme {
+                canonicalize_inside_type_scheme(type_scheme, path, paths);
+            }
+            if let Some(e) = e {
+                canonicalize_inside_expression(e, path, paths);
+            }
+        }
+        PilStatement::EnumDeclaration(_, enum_decl) => {
+            let type_vars = enum_decl.type_vars.vars().collect();
+            for variant in &mut enum_decl.variants {
+                if let Some(fields) = &mut variant.fields {
+                    for field in fields {
+                        canonicalize_inside_type(field, &type_vars, path, paths);
+                    }
+                }
+            }
+        }
+        PilStatement::TraitImplementation(_, trait_impl) => {
+            for f in &mut trait_impl.functions {
```

Comment:
Should also call `canonicalize_inside_type_scheme` for `trait_impl.type_scheme`.

Reply:
Extracted to https://github.com/powdr-labs/powdr/pull/1750/files because I'm not sure

Reply:
Following discussion, removed the commit and instead used generics on the existing data structures